Cinema system/Cinema system/Services/CinemaService.cs

```diff
@@ -81,5 +83,105 @@ public void DeleteCinema(int cinemaId)
 
             _db.SaveChanges();
         }
+
+        public bool IsHallExists(int cinemaId, int hallId)
+        {
+            return _db.Halls.Any(h => (h.Id == hallId && h.CinemaId == cinemaId && !h.IsDeleted));
+        }
+
+        public bool IsHallExists(int cinemaId, HallView hall)
+        {
+            return _db.Halls.Any(h => (h.Name == hall.Name && h.CinemaId == cinemaId && !h.IsDeleted));
+        }
+
+        public int CreateHall(int cinemaId, HallView hall)
+        {
+            var newHall = new Hall
+            {
+                Id = hall.Id,
+                Name = hall.Name,
+                Seats = hall.Seats
+                    .Select(s => new Seat
+                    {
+                        TypeOfSeat = (SeatType)Enum.Parse(typeof(SeatType), s.TypeOfSeat),
+                        Row = s.Row,
+                        Place = s.Place
+                    }).ToList()
+            };
+
+            var cinema = _db.Cinemas
+                 .Include(c => c.Halls)
+                 .FirstOrDefault(c => c.Id == cinemaId);
+
+            cinema.Halls.Add(newHall);
+
+            _db.SaveChanges();
+
+            return newHall.Id;
+        }
+
+        public void EditHall(int hallId, HallView hall)
+        {
+            var editHall = _db.Halls
+                .Include(h => h.Seats)
+                .FirstOrDefault(h => h.Id == hallId);
+
+            editHall.Name = hall.Name;
+
+            editHall.Seats = hall.Seats
+                .Select(s => new Seat
+                {
+                    TypeOfSeat = (SeatType)System.Enum.Parse(typeof(SeatType), s.TypeOfSeat),
+                    Row = s.Row,
+                    Place = s.Place
+                }).ToList();
+
+            _db.SaveChanges();
+        }
+
+        public IEnumerable<HallView> GetHalls(int cinemaId)
+        {
+            return _db.Halls
+                .Where(h => (h.CinemaId == cinemaId && !h.IsDeleted))
+                .Select(h => new HallView
+                {
+                    Id = h.Id,
+                    Name = h.Name,
+                    Seats = h.Seats
+                        .Select(s => new SeatView
+                        {
+                            TypeOfSeat = s.TypeOfSeat.ToString(),
+                            Row = s.Row,
+                            Place = s.Place
+                        })
```

Comment:
Do you really need all the seats for each hall when getting list of halls? What could be the use case for that?